src/omnicore/tx.cpp

```diff
@@ -1869,6 +1881,9 @@ int CMPTransaction::logicMath_RevokeTokens()
     assert(update_tally_map(sender, property, -nValue, BALANCE));
     assert(_my_sps->updateSP(property, sp));
 
+    // Number of tokens has changed, update fee distribution thresholds
+    p_feecache->UpdateDistributionThresholds();
```

Comment:
Just took a look at this, doesn't seem `updateSP` is the right place for the trigger - the class for a property (`CMSPInfo`) doesn't actually contain how many tokens exist for that property (and thus we can't rely on `updateSP` being called universally when token counts change).

An example of this is crowdsale purchases update the crowd database but `updateSP` isn't called until the crowdsale closes.